src/lib.rs

```diff
@@ -4892,13 +4904,14 @@ where
     #[inline]
     pub fn new_from_prefix(bytes: B) -> Result<(Ref<B, T>, B), CastError<B, T>> {
         util::assert_dst_is_not_zst::<T>();
-        let split_at = match Ptr::from_ref(bytes.deref()).try_cast_into::<T>(CastType::Prefix) {
-            Ok((_, split_at)) => split_at,
+        let remainder = match Ptr::from_ref(bytes.deref()).try_cast_into::<T>(CastType::Prefix) {
+            Ok((_, remainder)) => remainder,
             Err(e) => {
                 return Err(e.with_src(()).with_src(bytes));
             }
         };
 
+        let split_at = bytes.len().checked_sub(remainder.len()).unwrap();
```

Comment:
Heads up about this `.expect()`. If we go the "always return an error instead of panicking" route, even for errors we know are unreachable, then it's not clear to me what error to construct here (it was easier when we were returning `Option`s). I could also be convinced to do `unchecked_sub`.